query/query.go

```diff
@@ -122,6 +123,24 @@ type SubGraph struct {
 	Result []byte
 }
 
+var sgPool = sync.Pool{
+	New: func() interface{} {
+		return &SubGraph{}
+	},
+}
+
+func releaseSubgraph(root *SubGraph) {
+	if root == nil {
+		return
+	}
+
+	for _, child := range root.Children {
+		go releaseSubgraph(child)
```

Comment:
This will cost you more than the Pool will ever give back. Get rid of the `go` here and you should see a big improvement.